lib/internal/http2/compat.js

```diff
@@ -497,4 +522,37 @@ function onServerStream(stream, headers, flags) {
   server.emit('request', request, response);
 }
 
+class Http2PushRequest {
+  constructor(stream, headers) {
+    this[kStream] = stream;
+    this[kHeaders] = headers || Object.create(null);
+    this[kState] = {
+      headersSent: false
+    };
+  }
+
+  push(callback) {
+    var state = this[kState];
+    if (state.headersSent === true)
+      throw new Error('Cannot push request headers ' +
+        'after the HTTP message has been sent');
+    var stream = this[kStream];
+    if (stream === undefined)
+      throw new Error('HTTP/2 Stream has been closed');
+    var headers = this[kHeaders];
+    try {
+      stream.pushStream(
+        headers,
+        function(promisedStream, headers, options) {
+          var response = new Http2ServerResponse(promisedStream);
+          callback(null, response);
+        }
+      );
+      state.headersSent = true;
+    } catch (error) {
+      callback(error);
+    }
+  }
+}
```

Comment:
Would it be better to put all the logic in `createPushRequest()`?  Why do we need this `class`?

I promose we move the API from:

```js
const headers = {
   ':path': '/pushed',
   ':method': 'GET',
   ':scheme': 'http',
   ':authority': `localhost:${port}`
 };
 
const pushRequest = response.createPushRequest(headers);

pushRequest.push(common.mustCall(function(error, pushResponse) {
}))
```

to:

```js
const headers = {
   ':path': '/pushed',
   ':method': 'GET',
   ':scheme': 'http',
   ':authority': `localhost:${port}`
 };
 
response.createPushRequest(headers, common.mustCall(function(error, pushResponse) {
}))
```